### ISSUE TITLE
Turbopack assigning class variables as undefined explicitly on construction.

### ISSUE THREAD

**Issue by max-widmaier** [vercel/next.js#71891]
### Link to the code that reproduces this issue

https://github.com/max-widmaier/turbopack-explicit-undefined

### To Reproduce

1, Create a basic Next app
2. In src/app/page.tsx (jsx) use the following:

```typescript
class MyClass {
    public someProperty?: string;

    constructor() {
    }
}

export default function Page() {
    console.log(new MyClass());

    return <>Hello World</>
}
```

3. Run using `next dev`
4. Output will be
```javascript
MyClass {} // someProperty is implicitly undefined
```
5. Run using `next dev --turbo`
6. Output will be
```javascript
MyClass {
    someProperty: undefined // EXPLICIT UNDEFINED
}
```

### Current vs. Expected behavior

First for some background. I'm writing things with decorators to override get/set functionality of class properties. As part of this I've had to assign the property internally to `~${propertyName}` 
So assigning myObj.someProperty = 1234 and then console.logging myObj would give you:

```json
{
  "~someProperty": 1234
}
```

However, if "someProperty" is _explicity_ set to undefined, then the getter is overriden to get that undefined value. Normally, this is impossible because of the setter being overriden.

Doing otherwise leads to stackoverflows (assigning and then assigning and so on).

So, say I have my class:
```typescript
export class MyClass {
   @getSetOverride public someProperty?: number | null; 
}
```

Using the following code in a basic Next using Turbopack results in some unintended functionality (I used a page.tsx in the app router):

```typescript
export default function MyPage() {
   const myObj = new MyClass();
   myObj.someProperty = 1234;
   console.log(myObj);
}
```

The log results in:
```javascript
{
    someProperty: undefined,
    "~someProperty": 1234
}
```

Running `myObj.someProperty` then gives `undefined`.

Running the same code without Turbopack gives the proper value and the output is

```json
{
  "~someProperty": 1234
}
```

My assumption is that Turbopack is explicitly assigning class variables as undefined rather leaving them implicitly undefined.

You know you're doing something obscure when the build tool causes bugs...

TL;DR I would class instantiation to leave class field _implicitly_ undefined rather than _explicitly_ undefined. I have not assigned the fields to undefined so those fields shouldn't be touched.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Home
  Available memory (MB): 32666
  Available CPU cores: 16
Binaries:
  Node: 20.9.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.1 // Latest available version is detected (15.0.1).
  eslint-config-next: 14.2.13
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.6.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Turbopack

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

Tested on Next 14.2.13 and Next 15.0.1

**Comment by kdy1** [comment]
Currently, turbopack follows the specification. Old tools typically provide an option to ignore the specification, but turbopack defaults to the behavior declared in the specification.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#the-usedefineforclassfields-flag-and-the-declare-property-modifier
